gpkit/small_classes.py

```diff
@@ -35,13 +33,17 @@ def to_(self):  # used in tocoo, tocsc, etc below
     return to_
 
 
-class CootMatrix(CootMatrixTuple):
+class CootMatrix(object):
     "A very simple sparse matrix representation."
-    def __init__(self, *args, **kwargs):
-        super(CootMatrix, self).__init__(*args, **kwargs)
+    def __init__(self, row, col, data):
+        self.row, self.col, self.data = row, col, data
         self.shape = [(max(self.row) + 1) if self.row else 0,
                       (max(self.col) + 1) if self.col else 0]
 
+    def __eq__(self, other):
+        return (self.row == other.row and self.col == other.col
+                and self.data == other.data and self.shape == other.shape)
```

Comment:
would it be marginally faster to compare shape first (fail fast)? Or are the vast majority of comparisons expected to be between objects with the same shape?